kaybee/internal/tasks/pull.go

```diff
@@ -12,28 +11,25 @@ import (
 // conflicts using a set of pre-defined policies.
 type PullTask struct {
 	BaseTask
-	sources []conf.Source
+	sources []conf.SourceV1
```

Comment:
This breaks the generic approach of having multiple versions for the concrete configuration as loaded from files versus the unique "internal" schema that is independent of the versions; here pull.go has become dependent on V1 and cannot function with V2 or Vx

Reply:
I agree, I think it's better to implement an interface for `Source`, something like:
```golang
type Source interface{
   somefunc()
}

type ConfigV1 struct {
  Sources []SourceV1
}

type ConfigV2 struct {
  Sources map[int]SourceV2
}
```
However, it's problematic because configV2 would be a `map[int]interface{}` (interpreted as `Source`) and configV1 would be a `[]interface{}` (interpreted as `Source`). That's why right now the workaround was to convert a `ConfigV2.Sources` to a `ConfigV1.Sources`:

```golang
// config_v2.go 31
func (c ConfigV2) GetSources() []SourceV1 {
	sources := []SourceV1{}
	for _, source := range c.Sources {
		sources = append(sources, SourceV1{
			Repo:   source.Repo,
			Branch: source.Branch,
			Signed: true,
			Rank:   0,
		})
	}
	return sources
}
```

Ultimately I think it would be better to add either an SourceIterator interface or perform a type switch in the Execute function.

Reply:
I would work with the principle that the Execute function should be agnostic of the fact that different configuration schemas exist.